Tests/New-RsFolder.Tests.ps1

```diff
@@ -1,11 +1,84 @@
-        New-RsFolder -Path / -FolderName $folderName
+        $folderPath = '/'
+        New-RsFolder -Path $folderPath -FolderName $folderName
         $folderList = Get-RsFolderContent -RsFolder /
         $folderCount = ($folderList | where name -eq $folderName).Count
         It "Should be a new folder" {
             $folderCount | Should Be 1
         }
     }
+
+    Context "Create a subfolder"{
+        # Create folder to create the path
+        $parentFolderName = 'SutParentFolder' + [guid]::NewGuid()
+        New-RsFolder -Path / -FolderName $parentFolderName
+        $folderPath = '/'+ $parentFolderName
+        # Search for the folder path existence 
+        $folderList = Get-RsFolderContent -RsFolder /
+        $parentfolderCount = ($folderList | where path -eq $folderPath).Count
+        # Section to test the New-RsFolder
+        $subFolderName = 'SutSubFolder' + [guid]::NewGuid()
+        New-RsFolder -Path $folderPath -FolderName $subFolderName 
+        # Test if the folder was created
+        $allFolderList = Get-RsFolderContent -RsFolder / -Recurse
+        $subFolderPath = $folderPath + '/' + $subFolderName
+        $subFolderCount = ($allFolderList | where path -eq $subFolderPath).Count
+        It "Should the parent folder"{
+            $parentFolderCount | Should be 1
+        }
+        It "Should the subfolder"{
+            $subFolderCount | Should be 1    
+        }
+    }
+
+     Context "Create a folder with proxy"{
+        # Declaring the parameters name, path and proxy
+        $folderName = 'SutFolderParameterProxy' + [guid]::NewGuid()
+        $folderPath = '/'
+        $proxy = New-RsWebServiceProxy 
+        # Creating the folder with the parameters name, path and proxy
+        New-RsFolder -Path $folderPath -FolderName $folderName -Proxy $proxy
+        # Test if the folder was created
+        $folderList = Get-RsFolderContent -RsFolder /
+        $folderCount = ($folderList | where name -eq $folderName).Count
+        It "Should be a new folder with the parameter proxy"{
+            $folderCount | Should be 1    
+        }
+    }
+
+    Context "Create a folder with ReportServerUri"{
+        # Declaring the parameters name, path and ReportServerUri
+        $folderName = 'SutFolderParameterReportServerUri' + [guid]::NewGuid()
+        $folderPath = '/'
+        $folderReportServerUri = 'http://localhost/reportserver'
+        # Creating the folder with the parameters name, path, ReportServerUri
+        New-RsFolder -ReportServerUri $folderReportServerUri -Path $folderPath -FolderName $folderName
+        # Test if the folder was created
+        $folderList = Get-RsFolderContent -RsFolder /
+        $folderCount = ($folderList | where name -eq $folderName).Count
+        It "Should be a new folder with the parameter ReportServerUri" {
+            $folderCount | Should Be 1
+        }
+    }
+    
+    Context "Create a folder with all the parameters except credentials"{
```

Comment:
Discussed with @parthsha we should keep it as it test a behavior that we haven't designed yet, so it will work for the future